Cpp/source/Drawable/Drawable.hpp

```diff
@@ -0,0 +1,27 @@
+#pragma once
+#include "../FrameBuffer/FrameBuffer.hpp"
+#include <vector>
+#include <memory>
+
+struct GridPos
+{
+    int posX;
+    int posY;
+};
+
+class Drawable
+{
+protected:
+    GridPos position;
+    FrameBuffer frame;
+    std::vector<std::unique_ptr<Drawable>> drawables{};
```

Comment:
i'd say std::vector<std::shared_ptr < const Drawable > > this way whoever adds the drawables to drawable is sure it won't be modified.

Reply:
still applies